### DISCUSSION TITLE
wsl --shutdown and restart and DistroLauncher change group permissions on /run/user

### DISCUSSION THREAD

**Discussion by D3vil0p3r** [microsoft/WSL#9843] [General]
### Windows Version

Microsoft Windows [Version 10.0.22621.1265]

### WSL Version

1.1.3.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

Linux version 5.15.90.1-microsoft-standard-WSL2 (oe-user@oe-host) (x86_64-msft-linux-gcc (GCC) 9.3.0, GNU ld (GNU Binutils) 2.34.0.20200220) #1 SMP Fri Jan 27 02:56:13 UTC 2023

### Distro Version

Arch Linux latest

### Repro Steps

I'm testing an Arch Linux WSL image with GNOME 43 installed and I noted a strange behavior on WSL. I'm not sure if it occurs for all distros, but in this specific case, when you install a WSL image, let's guess Arch-based with GNOME, after the installation and the creation of the first user, that we can name `athena`, the `/run/user/1000` directory has the following permission details:
```
ll /run/user/1000
drwx------ athena 1000  180 B Fri Mar 24 11:37:43 2023  ./
drwxr-xr-x root   root   80 B Fri Mar 24 11:37:43 2023  ../
srw-rw-rw- athena users   0 B Fri Mar 24 11:37:43 2023  bus=
drwx------ athena users  60 B Fri Mar 24 11:37:43 2023  dbus-1/
drwx------ athena users  60 B Fri Mar 24 11:37:43 2023  gcr/
drwx------ athena users 140 B Fri Mar 24 11:37:43 2023  gnupg/
drwx------ athena users  60 B Fri Mar 24 11:37:43 2023  keyring/
drwxr-xr-x athena users  60 B Fri Mar 24 11:37:43 2023  pulse/
drwxr-xr-x athena users 140 B Fri Mar 24 11:37:43 2023  systemd/
```
Now, let's exit from the WSL session by `exit` command and come back to Windows CMD. Shutdown WSL by `wsl --shutdown` and then, re-enter in the WSL image. Now, if we go to see the details of the `/run/user/1000`, some permission details changed, and we get something like:
```
drwx------ athena 1000  220 B Fri Mar 24 11:39:07 2023  ./
drwxr-xr-x root   root   60 B Fri Mar 24 11:39:06 2023  ../
srw-rw-rw- athena users   0 B Fri Mar 24 11:39:06 2023  bus=
drwx------ athena 1000   60 B Fri Mar 24 11:39:04 2023  dbus-1/
drwx------ athena users  60 B Fri Mar 24 11:39:06 2023  gcr/
drwx------ athena users 140 B Fri Mar 24 11:39:06 2023  gnupg/
drwx------ athena users  60 B Fri Mar 24 11:39:06 2023  keyring/
drwx------ athena 1000   80 B Fri Mar 24 11:39:06 2023  pulse/
drwxr-xr-x athena users 140 B Fri Mar 24 11:39:06 2023  systemd/
srwxrwxrwx athena 1000    0 B Fri Mar 24 11:39:04 2023  wayland-0=
.rw-rw---- athena 1000    0 B Fri Mar 24 11:39:04 2023  wayland-0.lock
```
So, `wsl --shutdown` and then the WSL start will change some group permissions to `1000`, and generate two additional files: `wayland-0=` and `wayland-0.lock`. Is it possible to prevent that the group ownership on some files in `/run/user/1000` are changed when WSL is shutdown and restarted?
It should occur also when you set a different DEFAULT user by the Microsoft DistroLauncher.

### Expected Behavior

No group permission change on `/run/user/1000` at the restart of WSL.

### Actual Behavior

Group permission change to `1000` for some files on `/run/user/1000` at the restart of WSL.

**Comment by elsaco** [comment]
@D3vil0p3r what is the output of `id`? This is output on `Ubuntu-Preview` and WSL `1.1.5.0`:

```
elsaco@RIPPER:~$ ls -la /run/user/1000
total 0
drwx------ 6 elsaco elsaco 220 Mar 24 08:57 .
drwxr-xr-x 3 root   root    60 Mar 24 08:57 ..
srw-rw-rw- 1 elsaco elsaco   0 Mar 24 08:57 bus
drwx------ 3 elsaco elsaco  60 Mar 24 08:57 dbus-1
drwx------ 2 elsaco elsaco 140 Mar 24 08:57 gnupg
srw-rw-rw- 1 elsaco elsaco   0 Mar 24 08:57 pk-debconf-socket
drwx------ 2 elsaco elsaco  80 Mar 24 08:57 pulse
srw-rw-rw- 1 elsaco elsaco   0 Mar 24 08:57 snapd-session-agent.socket
drwxr-xr-x 5 elsaco elsaco 140 Mar 24 08:57 systemd
srwxrwxrwx 1 elsaco elsaco   0 Mar 24 08:57 wayland-0
-rw-rw---- 1 elsaco elsaco   0 Mar 24 08:57 wayland-0.lock
```

and real and effective user and group IDs for `elsaco`:

```
elsaco@RIPPER:~$ id
uid=1000(elsaco) gid=1000(elsaco) groups=1000(elsaco),4(adm),24(cdrom),27(sudo),30(dip),46(plugdev)
```

The output of `/run/user/1000` doesn't change after restarts.

**Comment by D3vil0p3r** [comment]
The output of `id` is:
```
uid=1000(athena) gid=984(users) groups=984(users),3(sys),90(network),98(power),991(lp),998(wheel)
```

**Comment by D3vil0p3r** [comment]
Is `elsaco` the default user in your WSL image? Could you try to create a new user and change this new one as the default one?

**Comment by benhillis** [comment]
/run is tmpfs. It is recreated each time the distro is launched.

**Comment by elsaco** [comment]
@D3vil0p3r it depends how you create users. After installing a fresh ArchLinux, first I created group id `1000` then add user `elsaco` to this group:

```
groupadd -g 1000 elsaco
useradd elsaco -u 1000 -g 1000 -m -s /bin/bash
```

and after restart this is the output of `/run/user/1000`:

```
[elsaco@RIPPER ~]$ ls -la /run/user/1000
total 0
drwx------ 7 elsaco elsaco 200 Mar 24 19:16 .
drwxr-xr-x 3 root   root    60 Mar 24 19:16 ..
srw-rw-rw- 1 elsaco elsaco   0 Mar 24 19:16 bus
drwx------ 3 elsaco elsaco  60 Mar 24 19:16 dbus-1
drwx------ 2 elsaco elsaco 140 Mar 24 19:16 gnupg
drwxr-xr-x 2 elsaco elsaco  60 Mar 24 19:16 p11-kit
drwx------ 2 elsaco elsaco  80 Mar 24 19:16 pulse
drwxr-xr-x 4 elsaco elsaco 120 Mar 24 19:16 systemd
srwxrwxrwx 1 elsaco elsaco   0 Mar 24 19:16 wayland-0
-rw-rw---- 1 elsaco elsaco   0 Mar 24 19:16 wayland-0.lock

```
and this is using the numeric UID argument:

```
[elsaco@RIPPER ~]$ ls -n /run/user/1000
total 0
srw-rw-rw- 1 1000 1000   0 Mar 24 19:16 bus
drwx------ 3 1000 1000  60 Mar 24 19:16 dbus-1
drwx------ 2 1000 1000 140 Mar 24 19:16 gnupg
drwxr-xr-x 2 1000 1000  60 Mar 24 19:16 p11-kit
drwx------ 2 1000 1000  80 Mar 24 19:16 pulse
drwxr-xr-x 4 1000 1000 120 Mar 24 19:16 systemd
srwxrwxrwx 1 1000 1000   0 Mar 24 19:16 wayland-0
-rw-rw---- 1 1000 1000   0 Mar 24 19:16 wayland-0.lock

```
OS info:
```
[elsaco@RIPPER ~]$ cat /etc/os-release
NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://bugs.archlinux.org/"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo
```

Since Arch has no WSL installer adding uid=1000 ang gid=1000 keeps in line with Ubuntu which adds a user with the same IDs on installation.

**Comment by D3vil0p3r** [comment]
Uhm, in my case the user has been created by `/usr/sbin/useradd --shell /bin/zsh -g users -G lp,network,power,sys,wheel --badname -m -N <username>`.
